### PR TITLE
[UXIT-1574] - Events Page · Concluded Event Layout

### PR DESCRIPTION
**Pull request by mirhamasala**
## 📝 Description

Please include a summary of the changes. Provide context and motivation for the change, and describe what problem it solves.

- **Type:** New feature + Refactor

## 🛠️ Key Changes

- **Create RecapSection component**
- **Include Concluded Event tag for concluded events**
-**Include Watch Event Highlights CTA**
- **Refactor EventsSection**

### New Features and Enhancements:

* `src/app/events/[slug]/components/RecapSection.tsx`: Added a new `RecapSection` component for displaying event recap videos using the `YouTubeVideoEmbed` component. ([src/app/events/[slug]/components/RecapSection.tsxR1-R26](diffhunk://#diff-aaec3db57280f62c3f635f326cb843617fe6c8b0e6b524380f404fd8045ed98fR1-R26))

### Configuration and Schema Updates:

* [`public/admin/config.yml`](diffhunk://#diff-c4b7720f4813ef734de91022075db713c78b4d0566dd7a95f868d75a65935486R681): Added hints for YouTube video URLs and new fields for event recap YouTube embed and playlist URLs. [[1]](diffhunk://#diff-c4b7720f4813ef734de91022075db713c78b4d0566dd7a95f868d75a65935486R681) [[2]](diffhunk://#diff-c4b7720f4813ef734de91022075db713c78b4d0566dd7a95f868d75a65935486R877-R885)
* [`src/app/events/schemas/FrontMatterSchema.ts`](diffhunk://#diff-6d034c3589bfaf98ab5ffaf4da649a3d13d58e7a5cd77a21aa879340549db864L32-R32): Made the `title` field in the `luma-events-section` optional.

### Code Refactoring and Simplification:

* [`src/app/_components/YouTubeVideoEmbed.tsx`](diffhunk://#diff-eb8a44359af076049ed6dc49b01ee69f972bfd67698ed6dcaebc0015edaa39e0R1-R63): Renamed, refactored and moved `VideoSection` to `YouTubeVideoEmbed` component
* [`src/app/_hooks/useEventsCategory.ts`](diffhunk://#diff-2b2b2a8547d05148e56298ebac117a8035faf7da605ff15ffa64ad47adf2a14eL3-R8): Refactored the `useEventsCategory` hook to simplify the logic for filtering past events by using the new `isEventConcluded` utility function. Removed the `filterByPastEvents` function. [[1]](diffhunk://#diff-2b2b2a8547d05148e56298ebac117a8035faf7da605ff15ffa64ad47adf2a14eL3-R8) [[2]](diffhunk://#diff-2b2b2a8547d05148e56298ebac117a8035faf7da605ff15ffa64ad47adf2a14eL24-R24) [[3]](diffhunk://#diff-2b2b2a8547d05148e56298ebac117a8035faf7da605ff15ffa64ad47adf2a14eL46-L53)
* [`src/app/_schemas/YouTubeVideoMetadataSchema.ts`](diffhunk://#diff-acadc1ce9f4472acb86eac89a08ae4f91f4e9897b2d1312ea8af733f6d6cf529L3-R3): Renamed and exported the `YouTubeVideoMetadataSchema` from `src/app/ecosystem-explorer/[slug]/services/youtube.ts` and moved the `fetchYouTubeVideoMetadata` function to `src/app/_services/youtube.ts`. [[1]](diffhunk://#diff-acadc1ce9f4472acb86eac89a08ae4f91f4e9897b2d1312ea8af733f6d6cf529L3-R3) [[2]](diffhunk://#diff-acadc1ce9f4472acb86eac89a08ae4f91f4e9897b2d1312ea8af733f6d6cf529L18-L25) [[3]](diffhunk://#diff-5c0b32962ca5646c95e21e7f8422c408586ad80c190578c93f0008d407cfdd00R1-R9)
* Removed `variant=primary` from `Button` instances since `primary` is already the  `variant` default.

## 📌 To-Do Before Merging

- [ ] Remove test data commit

## 🧪 How to Test

Go to `fil.com/events/fil-bangkok-2024` and follow the testing comments in `fil-bangkok-2024.md` to see the different options.

## 📸 Screenshots

CleanShot 2024-10-17 at 01 36 13@2x

### CODE REVIEW

File: src/app/events/[slug]/components/EventsSection.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import type { Event } from '@/types/eventType'
```
Comment (CharlyMartin) [inline src/app/events/[slug]/components/EventsSection.tsx:1]: I get this error here, not sure why

Cannot find module '@/types/eventType' or its corresponding type declarations.ts(2307)

File: src/app/events/[slug]/components/EventsSection.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import { PageSection } from '@/components/PageSection'
+
+type EventsSectionProps = Pick<
+  NonNullable<Event['lumaEventsSection']>,
+  'title' | 'embedLink'
+>
```
Comment (CharlyMartin) [inline src/app/events/[slug]/components/EventsSection.tsx:8]: Looking at the zod schema, could this just be `NonNullable<Event['lumaEventsSection']>`?

File: src/app/events/[slug]/page.tsx [hunk 3]
```diff
@@ -64,30 +73,36 @@ export default function EventEntry({ params }: EventProps) {
...
-        <TagLabel borderColor="brand-100">
-          {getCategoryLabel({ collectionName: 'event_entries', category })}
-        </TagLabel>
+        <TagGroup
+          label={[
+            getCategoryLabel({ collectionName: 'event_entries', category }),
+            eventHasConcluded ? 'Concluded Event' : undefined,
+          ]}
+        />
         <PageHeader
```
Comment (CharlyMartin) [inline src/app/events/[slug]/page.tsx:82]: Would be nice to add `objectFit: 'cover'` to the `PageHeader`. I thought I did it as part of the pervious PR.

**Review by CharlyMartin — APPROVED** [review]
Good one Mirha! I didn't see anything blocking ✨

But there is this `Cannot find module '@/types/eventType'` error to fix before we can merge this PR.

File: src/app/events/[slug]/page.tsx (hunk 3 above)
Comment (mirhamasala) [reply src/app/events/[slug]/page.tsx:82]: Ah! Nice catch! I was wanting to ask about this! 🙌🏼

**Pull request merged**